supervisor/shared/filesystem.c

```diff
@@ -117,7 +119,8 @@ void filesystem_init(bool create_allowed, bool force_create) {
 void filesystem_flush(void) {
     // Reset interval before next flush.
     filesystem_flush_interval_ms = CIRCUITPY_FILESYSTEM_FLUSH_INTERVAL_MS;
-    supervisor_flash_flush();
+    // Don't keep caches because this is called when starting or stopping the VM.
+    supervisor_flash_flush(false);
```

Comment:
This looks like the only case where you don't keep the cache. So would it make sense to always keep the cache, and just have this routine release the cache explicitly? (Make `release_ram_cache()` not be static (and rename it to `supervisor_....` ?)

Reply:
Ok, updated. Please take another look.